website/addons/wiki/utils.py

```diff
@@ -154,3 +155,64 @@ def format_wiki_version(version, num_versions, allow_preview):
         raise InvalidVersionError
 
     return version
+
+def serialize_wiki_settings(user, node_ids):
```

Comment:
OK, thanks for the clarification. Have you done any testing with projects with lots of components (e.g. RP:P-scale projects)? This function loads up all of a node's components, so it's possible that it significantly slows down the settings page.